src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SHA3_384.cs

```diff
@@ -0,0 +1,325 @@
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="source" /> does not support reading.
+        /// </exception>
+        /// <exception cref="PlatformNotSupportedException">
+        /// The platform does not support SHA3-384.
+        /// </exception>
+        public static ValueTask<byte[]> HashDataAsync(Stream source, CancellationToken cancellationToken = default)
+        {
+            ArgumentNullException.ThrowIfNull(source);
+
+            if (!source.CanRead)
+                throw new ArgumentException(SR.Argument_StreamNotReadable, nameof(source));
+
+            CheckSha3Support();
+            return LiteHashProvider.HashStreamAsync(HashAlgorithmNames.SHA3_384, source, cancellationToken);
+        }
+
+        /// <summary>
+        /// Asynchronously computes the hash of a stream using the SHA3-384 algorithm.
+        /// </summary>
+        /// <param name="source">The stream to hash.</param>
+        /// <param name="destination">The buffer to receive the hash value.</param>
+        /// <param name="cancellationToken">
+        ///   The token to monitor for cancellation requests.
+        ///   The default value is <see cref="System.Threading.CancellationToken.None" />.
+        /// </param>
+        /// <returns>The total number of bytes written to <paramref name="destination" />.</returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="source" /> is <see langword="null" />.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <para>
+        ///   The buffer in <paramref name="destination"/> is too small to hold the calculated hash
+        ///   size. The SHA3-384 algorithm always produces a 384-bit hash, or 48 bytes.
+        ///   </para>
+        ///   <para>-or-</para>
+        ///   <para>
+        ///   <paramref name="source" /> does not support reading.
+        ///   </para>
+        /// </exception>
+        /// <exception cref="PlatformNotSupportedException">
+        /// The platform does not support SHA3-384.
+        /// </exception>
+        public static ValueTask<int> HashDataAsync(
+            Stream source,
+            Memory<byte> destination,
+            CancellationToken cancellationToken = default)
+        {
+            ArgumentNullException.ThrowIfNull(source);
+
+            if (destination.Length < HashSizeInBytes)
+                throw new ArgumentException(SR.Argument_DestinationTooShort, nameof(destination));
+
+            if (!source.CanRead)
+                throw new ArgumentException(SR.Argument_StreamNotReadable, nameof(source));
+
+            CheckSha3Support();
+            return LiteHashProvider.HashStreamAsync(
+                HashAlgorithmNames.SHA3_384,
+                source,
+                destination,
+                cancellationToken);
+        }
+
+        private static void CheckSha3Support()
+        {
+            if (!IsSupported)
```

Comment:
Looking at things like this, per the previous discussion about these IsSupported properties caching the result via an implicit static readonly, I think they should.  Doing so will enable all of these CheckSha3Support methods to evaporate at JIT time on platforms that support it.